Userland/Libraries/LibWeb/Layout/Viewport.cpp

```diff
@@ -27,4 +27,55 @@ JS::GCPtr<Painting::Paintable> Viewport::create_paintable() const
     return Painting::ViewportPaintable::create(*this);
 }
 
+Vector<Viewport::TextBlock> Viewport::text_blocks()
+{
+    if (!m_text_blocks.has_value())
+        update_text_blocks();
+
+    return *m_text_blocks;
+}
+
+void Viewport::update_text_blocks()
+{
+    StringBuilder builder;
+    size_t current_start_position = 0;
+    Vector<TextPosition> text_positions;
+    Vector<TextBlock> text_blocks;
+    for_each_in_inclusive_subtree([&](auto const& layout_node) {
+        if (layout_node.display().is_none() || !layout_node.paintable() || !layout_node.paintable()->is_visible())
+            return TraversalDecision::Continue;
+
+        if (layout_node.is_block_container()) {
```

Comment:
hmm, if block container interrupts TextBlock then shouldn't the same happen for grid, flex and table containers?

Reply:
I've changed the check here to be `is_box()` instead. I *think* this covers all scenarios where we'd want a split to occur.